test/built-ins/Promise/prototype/finally/then-catch-finally-realm.js

```diff
@@ -0,0 +1,29 @@
+// Copyright (C) 2020 Tooru Fujisawa. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+description: >
+  Promise.prototype.finally creates thenFinally and catchFinally in onFinally's
+  realm
+esid: sec-promise.prototype.finally
+info: |
+  Promise.prototype.finally ( _onFinally_ )
+  6. Else,
+    a. Let realm be ? GetFunctionRealm(_onFinally_).
+    c. Let _thenFinally_ be ! CreateBuiltinFunction(_stepsThenFinally_, &laquo; [[Constructor]], [[OnFinally]] &raquo;, realm).
+    g. Let _catchFinally_ be ! CreateBuiltinFunction(_stepsCatchFinally_, &laquo; [[Constructor]], [[OnFinally]] &raquo;, realm).
+  7. Return ? Invoke(_promise_, *"then"*, &laquo; _thenFinally_, _catchFinally_ &raquo;).
+features: [Promise.prototype.finally]
+---*/
+
+var other = $262.createRealm().global;
+
+var onFinally = () => {};
```

Comment:
This test would be more precise if it used the "other" realm to create the `onFinally` function rather than using it to call `Promise.prototype.finally.` That kind of precision would reduce the risk of false positives (e.g. if the runtime incorrectly used the realm of the `then` function) and also give readers a stronger indication of the expected semantics (almost to say, "it should use the realm from this value, right here!").